src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -939,6 +948,81 @@ async function webviewPreloads(ctx: PreloadContext) {
 		}
 	}
 
+	function extractSelectionLine(selection: Selection): ISearchPreviewInfo {
```

Comment:
@rebornix 
> extractSelectionLine can be costly, I wonder if we want to request this info conditionally (when requested from the search widget). Also we might want to handle the case where we have multiple search results on the same line, which we should optimize the DOM read operations to 1.

> oh actually it seems to run extractSelectionLine all the time, can we make sure that it's also behind the experimental flag?

https://github.com/microsoft/vscode/pull/167952#discussion_r1100843845

Reply:
all callers of this function should check `shouldGetSearchPreviewInfo` is `True`. It shouldn't be run unless it needs to.